pints/_abc/_abc_smc.py

```diff
@@ -0,0 +1,198 @@
+        & \theta^{**} \sim K(\theta|\theta^{*}), \textrm{i.e. perturb }
+        \theta^{*} \textrm{ to     obtain to new point } x \sim
+        p(x|\theta^{**})\textrm{, i.e. sample data from sampling
+        distribution} \\
+        & \textrm{if } s(x) < \textrm{threshold}_(t), \theta^* \textrm{
+        added to list of samples[t]}
+        \end{align}
+
+    After we have obtained nr_samples samples, t is advanced, and weights
+    are calculated for samples[t-1]. At the last error threshold,
+    samples are returned whenever they are accepted. This algorithm is
+    also referred to as ABC Population Monte Carlo (ABC PMC) [2].
+
+    References
+    ----------
+    .. [1] "Toni, Tina, et al. Approximate Bayesian computation scheme
+            for parameter inference and model selection in dynamical systems.
+            Journal of the Royal Society Interface, 6.31: 187-202, 2009.
+            https://doi.org/10.1098/rsif.2008.0172
+
+    .. [2] "Beaumont, Mark A., et al. Adaptive approximate Bayesian
+            computation. Biometrika, 96.4: 983-990, 2009."
+            https://doi.org/10.48550/arXiv.0805.2256
+
+    Parameters
+    ----------
+    nr_samples
+        The number of samples requested for intermediate distributions.
+    error_schedule
+        The schedule of error threshold distance for all distributions.
+    """
+
+    def __init__(self, log_prior, perturbation_kernel=None):
+        # Log prior
+        self._log_prior = log_prior
+
+        # Default value for error threshold schedule
+        self._e_schedule = [1]
+
+        # Default value for current threshold
+        self._threshold = 1
+
+        # Size of intermediate distributions
+        self._nr_samples = 100
+
+        # Set up for first iteration
+        self._samples = [[]]
+        self._accepted_count = 0
+        self._weights = []
+        self._xs = None
+        self._ready_for_tell = False
+        self._t = 0
+        dim = log_prior.n_parameters()
+
+        # Setting the perturbation kernel
+        if perturbation_kernel is None:
+            self._perturbation_kernel = pints.MultivariateGaussianLogPrior(
+                np.zeros(dim),
+                0.001 * np.identity(dim))
+        elif isinstance(perturbation_kernel, pints.LogPrior):
+            self._perturbation_kernel = perturbation_kernel
+        else:
+            raise ValueError('Provided perturbation kernel must be an instance'
+                  ' of pints.LogPrior')
+
+    def name(self):
+        """ See :meth:`pints.ABCSampler.name()`. """
+        return 'ABC-SMC'
+
+    def ask(self, n_samples):
+        """ See :meth:`ABCSampler.ask()`. """
+        if self._ready_for_tell:
+            raise RuntimeError('ask called before tell.')
+        if self._t == 0:
+            self._xs = self._log_prior.sample(n_samples)
+        else:
+            self._xs = []
+
+            while len(self._xs) < n_samples:
+                theta_s_s = None  # to appease the linter
```

Comment:
```suggestion
                theta_s_s = None
```
The linter is right. Don't just use a variable you haven't defined yet :D